Data Snippets

ChefAndy opened the pull request:
This is just the back-end. The update_snippets fab task could be run periodically. Weekly? When more tasks are added it probably makes sense to parallelize them.

ChefAndy committed:
new fab task for updating snippets

ChefAndy committed:
forgot import

ChefAndy committed:
new model and migration

ChefAndy committed:
made new templatetag for snippets

ChefAndy committed:
template tag typo

ChefAndy committed:
new snippets view

ChefAndy committed:
new snippets url

ChefAndy committed:
added imports to views

jcushman commented on capstone/fabfile.py:
```diff
@@ -804,3 +804,106 @@ def report_multiple_jurisdictions(out_path="court_jurisdictions.csv"):
+    }
+
+    tally = {}
+    for volume in VolumeMetadata.objects.prefetch_related("case_metadatas__jurisdiction", 'volume_xml__page_xmls').all():
```
I think this loop isn't going to work for a couple of reasons. The `prefetch_related` is going to try to pull hundreds of gigabytes of case metadata and page xml into RAM, which will fail. The inner loop over cases also doesn't get the right answers -- for example 'volume_count' ends up as the number of cases, and 'case_count' ends up as the number of cases squared.

I think instead we want to do this as an aggregate sql query, or a few of them, on the capdb_jurisdiction table. Could either use a django Aggregate query (which are always frustrating) or a raw sql call. Happy to chat about this if it's helpful ...

ChefAndy replied:
Ah, good point. I'll change that.

jcushman reviewed (dismissed):
Nice! I like the Django view + templatetag approach. There's some issues with the `update_map_numbers` task, which isn't central to the Snippets work -- let me know if you want to fix the task, or mark it as a work in progress and pull as-is.

jcushman commented:
Oh, don't forget to rebase and recreate the migration, since it conflicts with your other PR that I pulled in.

ChefAndy commented:
already done

ChefAndy committed:
moved migration to avoid conflict

ChefAndy committed:
updated snippet migration to have the proper dependency

ChefAndy committed:
Merge remote-tracking branch 'upstream/develop' into new_branch_snippets

ChefAndy committed:
added and moved snippets scripts, made more efficient

snippet updates now in their own module
map update now uses celery to asychronously update in chunks of 1000
new snippet update scripts for TSVs

ChefAndy committed:
fixed some logic problems in my map numbers generator

ChefAndy committed:
added test, filtered out duplicative cases in reporter tsv

ChefAndy committed:
Merge remote-tracking branch 'upstream/develop' into new_branch_snippets

Conflicts: capstone/fabfile.py
-- messed up imports

ChefAndy committed:
made queries more efficient in update_map_numbers

ChefAndy commented:
updated

ChefAndy committed:
added search snippets and tests

ChefAndy committed:
Merge branch 'develop' of https://github.com/harvard-lil/capstone into new_branch_snippets

ChefAndy committed:
removed unnecessary arguments from templatetag
